extensions/notebook/src/contracts/content.ts

```diff
@@ -149,6 +149,5 @@ export interface IJupyterBookSectionV2 {
 	url?: string;
 }
 
-// type that supports new and old version
-export type JupyterBookSection = IJupyterBookSectionV1 | IJupyterBookSectionV2;
+export interface JupyterBookSection extends IJupyterBookSectionV1, IJupyterBookSectionV2 { }
```

Comment:
So when you edit a book is it going to write out the sections as the combined v1 & v2 section that you have internally? Is that normal/expected? Would be curious to know what people would think of that being done since from what I can tell it isn't obvious that's what would happen. How would other book related products handle that?

Reply:
That's expected, so internally we'll always use the v2 to avoid complexity. I combined them to keep the toc fields that are allowed in v1, but do not have an equivalent in v2.
Since at then I'm modifying the toc file, I don't want to miss any of those fields. Also before writing I check the version and if it's v1 I convert them.